src/unix/lwt_io.ml

```diff
@@ -1454,6 +1454,41 @@ let with_file ?buffer ?flags ?perm ~mode filename f =
     (fun () -> f ic)
     (fun () -> close ic)
 
+let open_temp_file_with_filename:
+  ?buffer : Lwt_bytes.t ->
+  ?perm : Unix.file_perm ->
+  unit -> (string * output_channel) Lwt.t
+  = fun ?buffer ?perm ->
+    let f = Some [Unix.O_WRONLY; Unix.O_CREAT; Unix.O_TRUNC;
+                  Unix.O_NONBLOCK; Unix.O_EXCL] in
```

Comment:
Nit, but I believe:

- `O_TRUNC` is not necessary, as you are passing `O_CREAT` and `O_EXCL`, so the file will not be an existing one.
- `O_NONBLOCK` has no effect on ordinary files, as being created here.

Reply:
It should also be possible to override the default flags. You usually want to add `O_CLOEXEC`, but not always ...